packages/xo-server/src/xapi/index.mjs

```diff
@@ -1053,25 +1053,33 @@ export default class Xapi extends XapiBase {
     } catch (error) {
       const { code } = error
       if (code === 'VDI_CBT_ENABLED') {
+        log.debug(`${vdi.name_label} has CBT enabled`)
         // 20240629 we need to disable CBT on all disks of the VM since the xapi
         // checks all disk of a VM even to migrate only one disk
-        if (vdi.VBDs.length === 0) {
+        const vbds = vdi.$VBDs.filter(({ $VM }) => $VM.is_control_domain === false)
+        if (vbds.length === 0) {
+          log.debug(`will disable CBT on ${vdi.name_label}  `)
           await this.call('VDI.disable_cbt', vdi.$ref)
         } else {
-          if (vdi.VBDs.length > 1) {
+          if (vbds.length > 1) {
             // no implicit workaround if vdi is multi attached
             throw error
           }
-          const vbd = this.getObject(vdi.VBDs[0])
+          const vbd = vbds[0]
+          log.debug(`will disable CBT on the full VM ${vbd.$VM.name_label}, containing disk ${vdi.name_label}  `)
           await this.VM_disableChangedBlockTracking(vbd.VM)
         }
 
         // cbt will be re enabled when needed on next backup
         // after a migration the next delta backup is always a base copy
         // and this will only enabled cbt on needed disks
 
-        // retry
-        return this.moveVdi(vdiId, srId, false)
+        // retry migrating disk
+        return this.barrier(
+          await pRetry(() => this.callAsync('VDI.pool_migrate', vdi.$ref, sr.$ref, {}), {
```

Comment:
check with guillaume if there is a way to have cbt_enable = true on a vdi but still raise a VDI_CBT_ENABLED error, maybe with metadata snapshot return this.moveVdi(vdiId, srId, false, { _failOnCbtError = true} )

if not, rewrite with

Reply:
`_failOnCbtError = false` and pass `true` on retry :wink: